test/IRCTest.h

```diff
@@ -188,6 +198,7 @@ class TestModule : public CModule {
     std::vector<CClient*> vClients;
     std::vector<CChan*> vChannels;
     EModRet eAction = CONTINUE;
+    bool bSendHooks = false;
```

Comment:
As an example of the sort of thing that would happen, [the test for `OnJoinMessage`][onjoinmessage] would have had to be changed from this:
```cpp
TEST_F(IRCSockTest, OnJoinMessage) {
    CMessage msg(":somebody JOIN #chan");
    m_pTestSock->ReadLine(msg.ToString());

    EXPECT_THAT(m_pTestModule->vsHooks, ElementsAre("OnJoinMessage"));
    EXPECT_THAT(m_pTestModule->vsMessages, ElementsAre(msg.ToString()));
    EXPECT_THAT(m_pTestModule->vNetworks, ElementsAre(m_pTestNetwork));
    EXPECT_THAT(m_pTestModule->vChannels, ElementsAre(m_pTestChan));
}
```
to this:
```cpp
TEST_F(IRCSockTest, OnJoinMessage) {
    CMessage msg(":somebody JOIN #chan");
    m_pTestSock->ReadLine(msg.ToString());

    EXPECT_THAT(m_pTestModule->vsHooks, ElementsAre("OnJoinMessage", "OnSendToClientMessage"));
    EXPECT_THAT(m_pTestModule->vsMessages, ElementsAre(msg.ToString(), msg.ToString()));
    EXPECT_THAT(m_pTestModule->vNetworks, ElementsAre(m_pTestNetwork, m_pTestNetwork));
    EXPECT_THAT(m_pTestModule->vChannels, ElementsAre(m_pTestChan, m_pTestChan));
}
```
Basically, it polluted many of the other tests. Worse, it wasn't super obvious when writing hook tests whether or not they would be affected.

The issue with `HALT` may have been something I was misremembering, as I can't find the relevant failed tests, but the idea is that, when testing some hook that also results in `OnSendToClient/IRCMessage` being called, if the hook action is set to `HALT`, it wasn't necessarily clear if the hook actually being tested would halt execution or if the `OnSend` hook would do so, resulting in incomplete testing.

By adding a field to indicate whether `OnSendToClient/IRCMessage` should process its calls, these issues were avoided.

[onjoinmessage]: https://github.com/elyscape/znc/blob/823ac07/test/IRCSockTest.cpp#L226-L234